R-package/R/lgb.Dataset.R

```diff
@@ -381,6 +385,12 @@ Dataset <- R6::R6Class(
       if (lgb.is.null.handle(x = private$handle)) {
         stop("Cannot get number of bins in feature before constructing Dataset.")
       }
+      if (is.character(feature)) {
+        feature <- which(colnames(self) == feature)
+        if (length(feature) == 0L) {
+          stop("feature not found")
+        }
```

Comment:
```suggestion
        feature_name <- feature
        feature <- which(colnames(self) == feature)
        if (length(feature) == 0L) {
          stop(sprintf("feature '%s' not found", feature_name))
        }
```

In my experience, including information about the input in error messages like this is really helpful for figuring out why you hit it. For example, if you misspelled a feature name that might be clearer when looking at an error message like `feature 'num_clickss' not found`.

Would you consider making this a bit more informative?

Reply:
added in fe6c9b4b6cf0f60af366dc402acaad9c182ceb31

Reply:
thanks very much!